File: android/common/src/main/java/com/innomalist/taxi/common/models/PaymentGateway.kt
package com.innomalist.taxi.common.models

import com.squareup.moshi.Json

data class PaymentGateway(
    var id: Int,
    var title: String?,
    var type: PaymentGatewayType,
    var publicKey: String? = null
)
enum class PaymentGatewayType (val rawValue: String) {
    @Json(name="stripe")
    Stripe("stripe"),
    @Json(name="braintree")
    Braintree("braintree"),
    @Json(name="flutterwave")
    Flutterwave("flutterwave");

    companion object {
        operator fun invoke(rawValue: String) = values().firstOrNull { it.rawValue == rawValue }
    }
}

File: ios/common/Models/Objects/PaymentGateway.swift
//
//  PaymentGateway.swift
//  Shared
//
//  Created by Manly Man on 12/11/19.
//  Copyright © 2019 Innomalist. All rights reserved.
//

import UIKit

public class PaymentGateway: Codable {
    public var id: Int
    var title: String
    public var type: PaymentGatewayType
    public var publicKey: String?
}

public enum PaymentGatewayType: String, Codable {
    case Stripe = "stripe"
    case Braintree = "braintree"
    case Flutterwave = "flutterwave"
}
